openstack/compute/v2/availabilityzones/utils.go

```diff
@@ -0,0 +1,35 @@
+package availabilityzones
+
+import (
+	"errors"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/availabilityzones"
+)
+
+// AZExists is a convienience function that checks if an availability zone exists and is
+// actually available.
+func AZExists(client *gophercloud.ServiceClient, name string) error {
+	allPages, err := availabilityzones.List(client).AllPages()
+	if err != nil {
+		return err
+	}
+
+	availabilityZoneInfo, err := availabilityzones.ExtractAvailabilityZones(allPages)
+	if err != nil {
+		return err
+	}
+	for _, zoneInfo := range availabilityZoneInfo {
+		if zoneInfo.ZoneName == name {
+			if zoneInfo.ZoneState.Available {
+				// AZ was found and is available
+				return nil
+			} else {
+				err := "AZ not available: " + zoneInfo.ZoneName
+				return errors.New(err)
+			}
+		}
+	}
+	notExist := "AZ does not exist: " + name
```

Comment:
On the caller side, how can you tell if the response is "doesn't exist" or if the error is actually caused by a critical failure? You'll probably have to type-assert the error, or worse pattern-match the error string.

If you want to keep the current behaviour, I'd change the return values to `(bool, error)`